src/main/java/org/minions/devfund/mauricioz/WhackAMole.java

```diff
@@ -0,0 +1,161 @@
+package org.minions.devfund.mauricioz;
+
+/**
+ * class in charge to provide game functionality.
+ */
+public class WhackAMole {
+    private static final int MOLES_LEFT = 10;
+    private int score = 0;
+    private int molesLeft;
+    private int attemptsLeft;
+    private char[][] moleGrid;
+
+    /**
+     * constructor.
+     * @param numAttempts number of attempts for the game.
+     * @param gridDimension grid size.
+     */
+    public WhackAMole(int numAttempts, int gridDimension) {
+        attemptsLeft = numAttempts;
+        moleGrid = new char[gridDimension][gridDimension];
+        this.molesLeft = MOLES_LEFT;
+        initializeMoleGrid();
+        placeMolesInTheGrid();
+    }
+
+    /**
+     * method to verify if whack was over a mole or not.
+     * @param x coordinate in x.
+     * @param y coordinate in y.
+     * @return true if mole was whacked.
+     */
+    public boolean place(int x, int y) {
+        if (moleGrid[x][y] == 'M') {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * print the initial grid.
+     */
+    public void printGridToUSer() {
+        for (int i = 0; i < moleGrid.length; i++) {
+            for (int j = 0; j < moleGrid[i].length; j++) {
+                System.out.print("\t * ");
+            }
+            System.out.println();
+        }
+    }
+
+    /**
+     * Whack over a mole.
+     * @param x coordinate in x.
+     * @param y coordinate in y.
+     */
+    public void whack(int x, int y) {
+        if (place(x, y)) {
+            moleGrid[x][y] = 'W';
+            molesLeft--;
+            score += 1;
+        }
+        attemptsLeft--;
+    }
+
+    /**
+     * print final grid.
+     */
+    public void printGrid() {
+        for (int i = 0; i < getMoleGridLength(); i++) {
+            for (int j = 0; j < moleGrid[i].length; j++) {
+                System.out.print(moleGrid[i][j] + "\t");
+            }
+            System.out.println();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)